test/dom/asyncHook.ts

```diff
@@ -1,5 +1,5 @@
 import { useState, useRef, useEffect } from 'react'
-import { renderHook } from '../src'
+import { renderHook } from '../../src/dom'
```

Comment:
I feel like I'm getting very close to moving the tests into `src/**__tests__/*` instead of the seperate root they're in now.

Reply:
My personal preference to have the tests as close to the code as possible, but I didn't want to change what you'd already done. If you're happy to do it that way, i'd prefer it.

Reply:
My preference for a seperate test root stems mostly from my Java background. I'm coming around to nested test directories and paths like this show the benefit.

This review is noisy enough with changes, so perhaps leave it for this PR and we can spin up a new one to move the tests once we're happy with the implementation (like you suggested for the docs)